lib/cli/src/commands/create_exe.rs

```diff
@@ -344,1043 +256,1624 @@ impl CreateExe {
+            let prefix_split = Self::split_prefix(p);
+
+            match prefix_split.as_slice() {
+                // ATOM:PREFIX:PATH
+                [atom, prefix, path] => {
+                    if only_validate_prefixes {
+                        // only insert the prefix in order to not error out of the fs::read(path)
+                        manual_prefixes.insert(atom.to_string(), prefix.to_string());
+                    } else {
+                        let atom_hash = atoms
+                        .iter()
+                        .find_map(|(name, _)| if name.as_str() == atom.as_str() { Some(prefix.to_string()) } else { None })
+                        .ok_or_else(|| anyhow::anyhow!("no atom {atom:?} found, for prefix {p:?}, available atoms are {available_atoms:?}"))?;
+
+                        let current_dir = std::env::current_dir().unwrap().canonicalize().unwrap();
+                        let path = current_dir.join(path.replace("./", ""));
+                        let bytes = std::fs::read(&path).map_err(|e| {
+                            anyhow::anyhow!("could not read file for atom {atom:?} (prefix {p}, path {} in dir {}): {e}", path.display(), current_dir.display())
+                        })?;
+
+                        compilation_objects.insert(atom.to_string(), bytes);
+                        manual_prefixes.insert(atom.to_string(), atom_hash.to_string());
+                    }
+                }
+                // atom + path, but default SHA256 prefix
+                [atom, path] => {
+                    let atom_hash = atoms
+                    .iter()
+                    .find_map(|(name, bytes)| if name.as_str() == atom.as_str() { Some(Self::hash_for_bytes(bytes)) } else { None })
+                    .ok_or_else(|| anyhow::anyhow!("no atom {atom:?} found, for prefix {p:?}, available atoms are {available_atoms:?}"))?;
+                    manual_prefixes.insert(atom.to_string(), atom_hash.to_string());
+
+                    if !only_validate_prefixes {
+                        let current_dir = std::env::current_dir().unwrap().canonicalize().unwrap();
+                        let path = current_dir.join(path.replace("./", ""));
+                        let bytes = std::fs::read(&path).map_err(|e| {
+                            anyhow::anyhow!("could not read file for atom {atom:?} (prefix {p}, path {} in dir {}): {e}", path.display(), current_dir.display())
+                        })?;
+                        compilation_objects.insert(atom.to_string(), bytes);
+                    }
+                }
+                // only prefix if atoms.len() == 1
+                [prefix] if atoms.len() == 1 => {
+                    manual_prefixes.insert(atoms[0].0.clone(), prefix.to_string());
+                }
+                _ => {
+                    return Err(anyhow::anyhow!("invalid --precompiled-atom {p:?} - correct format is ATOM:PREFIX:PATH or ATOM:PATH"));
+                }
```

Comment:
No, because it needs access to the `atoms` field, so it first needs to know what the atoms are (and for example, parse the pirita file to get all the valid atom names). So it's impossible to do this directly in clap, plus it would make it unnecessarily complicated just for the sake of having it in clap.